### DISCUSSION TITLE
Support for Amanon MQ on AWS

### DISCUSSION THREAD

**Discussion by adambezecny** [rabbitmq/rabbitmq-delayed-message-exchange#184] [General]
Hi,

is it possible to run this plughin on Amazon MQ? AWS documentation does not mention this: https://docs.aws.amazon.com/amazon-mq/latest/developer-guide/rabbitmq-basic-elements-plugins.html

I am wondering whether there is any workaround how to enable it

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
You have to ask AWS about that. This plugin works with vanilla open source RabbitMQ. I don't see why it would not work with whatever they run but we don't know what that is, or whether arbitrary plugins would ever be supported (for good reasons).

**Comment by sivudu** [comment]
Hi @adambezecny : We currently do not support delayed message exchange plugin on Amazon MQ. Sorry about the delayed response. I've not been monitoring the discussions here.

**Reply by yasinlachiny** [reply]
@adambezecny 
Do you have any plans to add it?

**Comment by yasinlachiny** [comment]
Is there any update about this issue?

**Comment by bmariesan** [comment]
@sivudu do you have any plans or scheduled roadmap to include the delayed message exchange plugin?

**Reply by sivudu** [reply]
Hi @bmariesan : I'm unable to offer a definitive timeline at this point but we are looking into enabling support for this plug-in. Until then, is it an option for you to configure time-to-live expiration on the queues where you need delayed delivery and consume messages off of a dead-letter exchange to mimic the basic behaviour of delayed message exchange plug-in?

**Comment by bmariesan** [comment]
@michaelklishin @sivudu since this discussion has been closed, has support for the delayed plugin been added to Amazon MQ? currently I don't see any update in the official docs

**Reply by sivudu** [reply]
Hi @bmariesan : Not yet. We are working with the community to improve HA properties of this plug-in.

**Reply by dvdtth** [reply]
any updates on making this plugin available in amazon mq?
curious to know if anyone has got this plugin working (somehow) in amazon mq, or what's the alternative approach to make fill this gap.

**Reply by michaelklishin** [reply]
@dvdtth please use an Amazon-specific forum to ask Amazon MQ-specific questions

**Reply by bmariesan** [reply]
@dvdtth we fixed this with 4 queues - the main queue, a delay queue with a TTL of let's say 5-10-15 seconds whatever you want, whenever the TTL expires the message is routed to the DLQ, for the DLQ we have a listener that adds/updates a retry header and posts the message to the main queue if retry count is lesser than what we want or to a parking lot queue if retries have been exhausted => in our case the parking lot requires manual intervention

**Comment by michaelklishin** [comment]
This discussion belongs to an AmazonMQ support forum.
